File: app/src/main/java/com/bkjcb/rqapplication/stationCheck/retrofit/ApplianceCheckService.java
package com.bkjcb.rqapplication.stationCheck.retrofit;

import com.bkjcb.rqapplication.stationCheck.model.ApplianceCheckResult;
import com.bkjcb.rqapplication.base.model.HttpResult;

import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by DengShuai on 2020/1/6.
 * Description :
 */
public interface ApplianceCheckService {
    @GET("/rq/push/getunitname")
    Observable<ApplianceCheckResult> getUnitName(@Query("type") int type);

    @GET("/rq/push/getFixDic")
    Observable<ApplianceCheckResult> getFixCheckItem();

    @GET("/rq/push/getReport")
    Observable<ApplianceCheckResult> getReportCheckItem();

    @GET("/rq/push/getSaleCheck")
    Observable<ApplianceCheckResult> getSaleCheckItem();

    @POST("/rq/push/saveDailyCheck")
    @FormUrlEncoded
    Observable<HttpResult> saveDailyCheck(
            @Field("guid") String guid,
            @Field("checkMan") String userId,
            @Field("unitGUID") String unitID,
            @Field("checktime2") String time,
            @Field("Opinions") String Opinions,
            @Field("guidArry") String zdjcxid,
            @Field("tableJson") String jianchajilu,
            @Field("phoneftp") String ftp
    );

    @POST("/rq/push/saveAlarmDailyCheck")
    @FormUrlEncoded
    Observable<HttpResult> saveAlarmDailyCheck(
            @Field("guid") String guid,
            @Field("checkMan") String userId,
            @Field("unitGUID") String unitID,
            @Field("checktime2") String time,
            @Field("Opinions") String Opinions,
            @Field("guidArry") String zdjcxid,
            @Field("tableJson") String jianchajilu,
            @Field("phoneftp") String ftp
    );

    @POST("/rq/push/saveSaleDailyCheck")
    @FormUrlEncoded
    Observable<HttpResult> saveSaleDailyCheck(
            @Field("guid") String guid,
            @Field("checkMan") String userId,
            @Field("unitGUID") String unitID,
            @Field("checktime2") String time,
            @Field("Opinions") String Opinions,
            @Field("guidArry") String zdjcxid,
            @Field("tableJson") String jianchajilu,
            @Field("phoneftp") String ftp
    );
}

File: app/src/main/java/com/bkjcb/rqapplication/check/retrofit/ApplianceCheckService.kt
package com.bkjcb.rqapplication.check.retrofit

import com.bkjcb.rqapplication.check.model.ApplianceCheckResult
import com.bkjcb.rqapplication.model.HttpResult
import io.reactivex.Observable
import retrofit2.http.*

/**
 * Created by DengShuai on 2020/1/6.
 * Description :
 */
interface ApplianceCheckService {
    @GET("/rq/push/getunitname")
    fun getUnitName(@Query("type") type: Int): Observable<ApplianceCheckResult>

    @GET("/rq/push/getFixDic")
    fun fixCheckItem(): Observable<ApplianceCheckResult>

    @GET("/rq/push/getReport")
    fun reportCheckItem(): Observable<ApplianceCheckResult>

    @GET("/rq/push/getSaleCheck")
    fun saleCheckItem(): Observable<ApplianceCheckResult>

    @POST("/rq/push/saveDailyCheck")
    @FormUrlEncoded
    fun saveDailyCheck(
            @Field("guid") guid: String?,
            @Field("checkMan") userId: String?,
            @Field("unitGUID") unitID: String?,
            @Field("checktime2") time: String?,
            @Field("Opinions") Opinions: String?,
            @Field("guidArry") zdjcxid: String?,
            @Field("tableJson") jianchajilu: String?,
            @Field("phoneftp") ftp: String?
    ): Observable<HttpResult>

    @POST("/rq/push/saveAlarmDailyCheck")
    @FormUrlEncoded
    fun saveAlarmDailyCheck(
            @Field("guid") guid: String?,
            @Field("checkMan") userId: String?,
            @Field("unitGUID") unitID: String?,
            @Field("checktime2") time: String?,
            @Field("Opinions") Opinions: String?,
            @Field("guidArry") zdjcxid: String?,
            @Field("tableJson") jianchajilu: String?,
            @Field("phoneftp") ftp: String?
    ): Observable<HttpResult>

    @POST("/rq/push/saveSaleDailyCheck")
    @FormUrlEncoded
    fun saveSaleDailyCheck(
            @Field("guid") guid: String?,
            @Field("checkMan") userId: String?,
            @Field("unitGUID") unitID: String?,
            @Field("checktime2") time: String?,
            @Field("Opinions") Opinions: String?,
            @Field("guidArry") zdjcxid: String?,
            @Field("tableJson") jianchajilu: String?,
            @Field("phoneftp") ftp: String?
    ): Observable<HttpResult>
}